diff --git a/lib/device/XRTrackedInput.js b/lib/device/XRTrackedInput.js
index c767f8d90ce42049d6c5db5320ca78f6766ee162..1cf696736f6e7865d42e6b84c7be177c49d765d6 100644
--- a/lib/device/XRTrackedInput.js
+++ b/lib/device/XRTrackedInput.js
@@ -67,10 +67,6 @@ export class XRTrackedInput {
                 if (button[P_GAMEPAD].eventTrigger != null) {
                     if (button[P_GAMEPAD].lastFrameValue === 0 &&
                         button[P_GAMEPAD].value > 0) {
-                        session.dispatchEvent(new XRInputSourceEvent(button[P_GAMEPAD].eventTrigger, {
-                            frame,
-                            inputSource: this[P_TRACKED_INPUT].inputSource,
-                        }));
                         session.dispatchEvent(new XRInputSourceEvent(button[P_GAMEPAD].eventTrigger + 'start', {
                             frame,
                             inputSource: this[P_TRACKED_INPUT].inputSource,
@@ -82,6 +78,10 @@ export class XRTrackedInput {
                             frame,
                             inputSource: this[P_TRACKED_INPUT].inputSource,
                         }));
+                        session.dispatchEvent(new XRInputSourceEvent(button[P_GAMEPAD].eventTrigger, {
+                            frame,
+                            inputSource: this[P_TRACKED_INPUT].inputSource,
+                        }));
                     }
                 }
             }
