diff --git a/build/three.webgpu.js b/build/three.webgpu.js
index 8cbb77a7bd36c6a89d397f18f8758fc60ff8146e..c1f87477640e683e447bcbf97645a64e8b5a22c6 100644
--- a/build/three.webgpu.js
+++ b/build/three.webgpu.js
@@ -14952,7 +14952,11 @@ const cameraPosition = /*@__PURE__*/ ( Fn( ( { camera } ) => {
 
 		if ( _cameraPositionArray === null ) {
 
-			_cameraPositionArray = uniformArray( positions ).setGroup( renderGroup ).setName( 'cameraPositions' ).onRenderUpdate( ( { camera }, self ) => {
+			_cameraPositionArray = uniformArray( positions ).setGroup( renderGroup ).setName( 'cameraPositions' ).onRenderUpdate( ( frame, self ) => {
+
+				if ( frame === undefined ) return;
+
+				const { camera } = frame;
 
 				const subCameras = camera.cameras;
 				const array = self.array;
@@ -71131,7 +71135,7 @@ class WebGLState {
 
 		let needsUpdate = false;
 
-		if ( renderContext.textures !== null ) {
+		if ( renderContext.textures !== null && framebuffer !== null ) {
 
 			drawBuffers = this.currentDrawbuffers.get( framebuffer );
 
diff --git a/build/three.webgpu.nodes.js b/build/three.webgpu.nodes.js
index fda563ffd328b99dc693d884f51fd39beb530a47..ee31416d454d94469465bd949b2d9ac0636b3266 100644
--- a/build/three.webgpu.nodes.js
+++ b/build/three.webgpu.nodes.js
@@ -14952,7 +14952,11 @@ const cameraPosition = /*@__PURE__*/ ( Fn( ( { camera } ) => {
 
 		if ( _cameraPositionArray === null ) {
 
-			_cameraPositionArray = uniformArray( positions ).setGroup( renderGroup ).setName( 'cameraPositions' ).onRenderUpdate( ( { camera }, self ) => {
+			_cameraPositionArray = uniformArray( positions ).setGroup( renderGroup ).setName( 'cameraPositions' ).onRenderUpdate( ( frame, self ) => {
+
+				if ( frame === undefined ) return;
+
+				const { camera } = frame;
 
 				const subCameras = camera.cameras;
 				const array = self.array;
@@ -71131,7 +71135,7 @@ class WebGLState {
 
 		let needsUpdate = false;
 
-		if ( renderContext.textures !== null ) {
+		if ( renderContext.textures !== null && framebuffer !== null ) {
 
 			drawBuffers = this.currentDrawbuffers.get( framebuffer );
 
diff --git a/src/nodes/accessors/Camera.js b/src/nodes/accessors/Camera.js
index 5e89cb93c61f37a1a2d03a27735c74287c0f0e9f..167c7226183d62af9ec61911a742e2445f58a5bc 100644
--- a/src/nodes/accessors/Camera.js
+++ b/src/nodes/accessors/Camera.js
@@ -315,7 +315,11 @@ export const cameraPosition = /*@__PURE__*/ ( Fn( ( { camera } ) => {
 
 		if ( _cameraPositionArray === null ) {
 
-			_cameraPositionArray = uniformArray( positions ).setGroup( renderGroup ).setName( 'cameraPositions' ).onRenderUpdate( ( { camera }, self ) => {
+			_cameraPositionArray = uniformArray( positions ).setGroup( renderGroup ).setName( 'cameraPositions' ).onRenderUpdate( ( frame, self ) => {
+
+				if ( frame === undefined ) return;
+
+				const { camera } = frame;
 
 				const subCameras = camera.cameras;
 				const array = self.array;
diff --git a/src/renderers/webgl-fallback/utils/WebGLState.js b/src/renderers/webgl-fallback/utils/WebGLState.js
index 897be18b97d0810929e0e60b7700ca1c0c81f547..3881fdca8c926129ee4787179957af13053db83b 100644
--- a/src/renderers/webgl-fallback/utils/WebGLState.js
+++ b/src/renderers/webgl-fallback/utils/WebGLState.js
@@ -1154,7 +1154,7 @@ class WebGLState {
 
 		let needsUpdate = false;
 
-		if ( renderContext.textures !== null ) {
+		if ( renderContext.textures !== null && framebuffer !== null ) {
 
 			drawBuffers = this.currentDrawbuffers.get( framebuffer );
 
